aries_cloudagent/protocols/did_rotate/v1_0/routes.py

```diff
@@ -0,0 +1,121 @@
+"""DID rotate admin routes."""
+
+import logging
+
+from aiohttp import web
+from aiohttp_apispec import docs, json_schema, match_info_schema, response_schema
+from marshmallow import fields
+
+from ....admin.request_context import AdminRequestContext
+from ....connections.models.conn_record import ConnRecord
+from ....messaging.models.openapi import OpenAPISchema
+from ....messaging.valid import DID_WEB_EXAMPLE, UUID4_EXAMPLE
+from ....storage.error import StorageNotFoundError
+from .manager import DIDRotateManager
+from .message_types import SPEC_URI
+from .messages.hangup import HangupSchema as HangupMessageSchema
+from .messages.rotate import RotateSchema as RotateMesageSchema
+
+LOGGER = logging.getLogger(__name__)
+
+
+class DIDRotateConnIdMatchInfoSchema(OpenAPISchema):
+    """Request to rotate a DID."""
+
+    conn_id = fields.String(
+        required=True,
+        metadata={"description": "Connection identifier", "example": UUID4_EXAMPLE},
+    )
+
+
+class DIDRotateRequestJSONSchema(OpenAPISchema):
+    """Request to rotate a DID."""
+
+    to_did = fields.String(
+        required=True,
+        metadata={
+            "description": "The DID the rotating party is rotating to",
+            "example": DID_WEB_EXAMPLE,
+        },
+    )
+
+
+@docs(tags=["did-rotate"], summary="Begin rotation of a DID as a rotator")
+@match_info_schema(DIDRotateConnIdMatchInfoSchema())
+@json_schema(DIDRotateRequestJSONSchema())
+@response_schema(
+    RotateMesageSchema(), 200, description="Rotate agent message for observer"
+)
+async def rotate(request: web.BaseRequest):
+    """Request to rotate a DID."""
+
+    LOGGER.debug("DID Rotate Rotate request >>>")
+
+    context: AdminRequestContext = request["context"]
+
+    profile = context.profile
+    did_rotate_mgr = DIDRotateManager(profile)
+
+    connection_id = request.match_info["conn_id"]
+
+    body = await request.json()
+    to_did = body["to_did"]
```

Comment:
This implies that controller must first create a new DID to rotate the connection to. For pairwise connections (likely just about every single connection), we'd want to generate a new DID for every rotation. If you're rotating in bulk, that's double the number of requests... If rotating in bulk is needed, perhaps it would be best for there to be a separate Admin API for that.

The other issue is I don't believe there's currently a way to create a did:peer:2 or 4 through the Admin API. Creation of did:peer:2/4 is only done in DID Exchange at the moment. It shouldn't be too crazy to add to the `/wallet/did/create` route, I think. If we don't implement that in this PR, I think we'd want it before a release because the DID rotate protocol would be basically useless for the most common case without it.

Reply:
I'd have to think about how to implement a bulk rotate in a nice interface given how the RFC is written. To be fair I have implemented the `/rotate` route directly based on how the message is described [here](https://github.com/hyperledger/aries-rfcs/tree/fe79f41e18786b628a91d65c1e13cfb824d2e16f/features/0794-did-rotate#rotate), not really thinking about how the DID would be generated in the first place. I think it makes sense to include a route (or update a current one) to accommodate the creation of did:peer:2/4 which would be fed into the `/rotate` endpoint. If we want to then follow up with a route to create and rotate to a new DID in one step, we can add a convenience handler for that.

Am I following you correctly?

Reply:
Yep, that's about what I had in mind. Though, to clarify, I don't think it's necessary to create that bulk endpoint now. I think the minimum adjustment required to merge this PR is the did:peer:2/4 creation.

Reply:
This is now implemented in the /wallet/did POST endpoint. Tests have been updated.